### ISSUE TITLE
Virtual & Physical Keyboard Does not Work For Typing While Playing Game

### ISSUE THREAD

**Issue by d10sfan** [ValveSoftware/SteamOS#1053]
#### Your system information

* Steam client version: 1685488080
* SteamOS version: 3.4.8
* Opted into Steam client beta?: [Yes/No]  No
* Opted into SteamOS beta?: [Yes/No]  No
* Have you checked for updates in Settings > System?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Many Linux native games (Like hl2 or running source ports like gzdoom) have an issue where typing with a physical or the on-screen keyboard fails to work. For example, opening the console and trying to type with a keyboard fails, but the keyboard works to control the game. It seems there may be some issue with steam input interfering with normal keyboard operations. I had opened an issue in the hl2 repo, but this seems to be more widespread/steamos related.

#### Steps for reproducing this issue:

1. Open a Linux native game like half-life 2
2. Open the developer console
3. Try to type with the on-screen keyboard or a virtual keyboard

**Comment by james2432** [comment]
could be related to:

https://github.com/ValveSoftware/SteamOS/issues/1027

**Comment by d10sfan** [comment]
In this case the keyboard is visible but does not work

**Comment by d10sfan** [comment]
This still happens on 3.5

**Comment by cmbernard333** [comment]
Seeing the same issue. Mostly in non-steam games.

**Comment by Blitzy** [comment]
I'm having this issue using a native Linux build of GZDoom through Luxtorpeda. I can't write in the console and I can't name save games. Obviously other keys work, my controller scheme is Keyboard + Mouse based, but the moment a text input is activated it seems like none of the alphanumeric keys register?

**Comment by cmbernard333** [comment]
I haven't checked in latest, but back in October this was my experience as well using an AppImage that launches the flatpak of GZDoom.

**Comment by cmbernard333** [comment]
Re-tested with latest and this is still an issue.

**Comment by TheOnlyPete** [comment]
I'm currently having this issue with Strife: Veteran Edition, you can't enter a file name and thus can't play the game

**Comment by bertogg** [comment]
This could be related to #1039

**Comment by d10sfan** [comment]
It doesn't apply only to zenity, also applies to in game typing as well

**Comment by cmbernard333** [comment]
Mine is an app image launcher that configures and launches the flatpak for gzdoom. I can see if I can configure the launcher itself to use the env var.

**Comment by bertogg** [comment]
Confirmed with Portal 2, so I suspect that it should also work with Half-Life 2.

I'm pretty sure that the problem is https://github.com/libsdl-org/SDL/issues/8561

**Comment by cmbernard333** [comment]
This worked for me using the same target dir and starting dir as the app image I was already using.

```shell
#!/usr/bin/env bash
export SteamDeck=0
./<app-image-path> --no-sandbox
```

**Comment by d10sfan** [comment]
I'll give that a shot thanks, setting that env variable off has also helped for other issues, like openmw having really low framerate. Are there any sort of downsides to doing this?

**Comment by bertogg** [comment]
> Are there any sort of downsides to doing this?

I'm not sure how many other apps or libs are checking the `SteamDeck` variable apart from libSDL. I would apply the changes locally to the games that are having problems while I wait for a fixed version of libSDL. If you ask in the libSDL issue they can probably give you more details, I'm not that familiar with how that library handles on-screen keyboards.

**Comment by cmbernard333** [comment]
@bertogg can you link the libSDL issue? I'd like to add some color as well.

**Comment by mikehgentry** [comment]
It's linked here:

> Confirmed with Portal 2 (in this case setting `SteamDeck=0 %command%` in the launch options), so I suspect that it should also work with Half-Life 2.
> 
> I'm pretty sure that the problem is [libsdl-org/SDL#8561](https://github.com/libsdl-org/SDL/issues/8561)

**Comment by bertogg** [comment]
FYI this problem has been fixed in libSDL 2.30.0, now it's just a matter of time before it reaches the Steam and Flatpak runtimes, etc.

I confirmed that the problem is solved with some of the games mentioned here but if anyone notices anything strange with those or other games please report it, thanks!
